src/window/building/distribution.c

```diff
@@ -194,41 +199,84 @@ static void draw_accept_none_button(int x, int y, int focused)
+        
+        if (!permissionState) {
+            inner_panel_draw(x + 2, y + 2, 3, 3);
+        }
+        image_offset_x = is_sea_trade_route ? 12 : 7;
+        image_offset_y = is_sea_trade_route ? 16 : 7;
+        
+        image_draw(images_permission[i], x + image_offset_x, y + image_offset_y, COLOR_MASK_NONE, SCALE_NONE);
+
+        if (!permissionState) {
+            image_draw(assets_get_image_id("UI", "Large_Widget_Cross"), x + 15, y + 15,
+            COLOR_MASK_NONE, SCALE_NONE);
         }
-        dx += offsets[i];
+        
+        button_border_draw(x, y, 52, 52, data.permission_focus_button_id == i + 1 || !permissionState ? 1 : 0);
+
+        x += 62;
     }
+
     building *b = building_get(c->building_id);
     int button = 1;
     if (building_storage_get_permission(BUILDING_STORAGE_PERMISSION_WORKER, b)) {
         button = 2;
     }
-    image_buttons_draw(x, y, image_buttons_maintain, button);
+    image_buttons_draw(c->x_offset + 421, c->y_offset + 10, image_buttons_maintain, button);
 }
 
 static void draw_granary_permissions_buttons(int x, int y, int buttons)
-{
-    int offsets[] = { 96, 96, 132, 96 };
+{   
+    int images_permission[] = {
+        assets_get_image_id("Walkers", "marketbuyer_sw_01"),        
+        assets_get_image_id("Walkers", "M Hall SW 01"),
+        image_group(GROUP_FIGURE_TRADE_CARAVAN) + 4,
+        image_group(GROUP_EMPIRE_TRADE_ROUTE_TYPE),
+        assets_get_image_id("Walkers", "Barkeep SW 01"),
+        assets_get_image_id("Walkers", "Slave SW 01"),
+    };
+
+    int image_offset_x, image_offset_y;
+
     for (int i = 0; i < buttons; i++) {
         int permission = granary_distribution_permissions_buttons[i].parameter1 - 1;
-        button_border_draw(x, y, 20, 20, data.permission_focus_button_id == i + 1 ? 1 : 0);
-        if (building_storage_get_permission(permission, building_get(data.building_id))) {
-            image_draw(assets_get_image_id("UI", "Allowed_Walker_Check"), x + 4, y + 4,
-                COLOR_MASK_NONE, SCALE_NONE);
+        int is_sea_trade_route = permission == BUILDING_STORAGE_PERMISSION_DOCK;
+        int permissionState = building_storage_get_permission(permission, building_get(data.building_id));
+        
+        if (!permissionState) {
+            inner_panel_draw(x + 2, y + 2, 3, 3);
         }
-        x += offsets[i];
+        image_offset_x = is_sea_trade_route ? 12 : 7;
+        image_offset_y = is_sea_trade_route ? 16 : 7;
+        
+        image_draw(images_permission[i], x + image_offset_x, y + image_offset_y, COLOR_MASK_NONE, SCALE_NONE);
+
+        if (!permissionState) {
+            image_draw(assets_get_image_id("UI", "Large_Widget_Cross"), x + 15, y + 15,
+            COLOR_MASK_NONE, SCALE_NONE);
+        }
+        
+        button_border_draw(x, y, 52, 52, data.permission_focus_button_id == i + 1 || !permissionState ? 1 : 0);
```

Comment:
```suggestion
        button_border_draw(x, y, 52, 52, data.permission_focus_button_id == i + 1 || !permission_state);
```

This also works and is simpler to read.